### PR TITLE
linuxPackages.yt6801: init at 1.0.29

### PR DESCRIPTION
**Pull request by 5aaee9**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/os-specific/linux/yt6801/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  version = "1.0.29";
+  src = fetchzip {
+    stripRoot = false;
+    url = "https://www.motor-comm.com/Public/Uploads/uploadfile/files/20240812/yt6801-linux-driver-${version}.zip";
+    sha256 = "sha256-oz6CeOUN6QWKXxe3WUZljhGDTFArsknjzBuQ4IchGeU=";
+  };
+
+  buildInputs = [ kmod ];
+  nativeBuildInputs = kernel.moduleBuildDependencies;
```
Comment (FliegendeWurst) [inline pkgs/os-specific/linux/yt6801/default.nix:24]:
```suggestion
  nativeBuildInputs = kernel.moduleBuildDependencies ++ [ kmod ];
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/yt6801/default.nix (hunk 1 above)
Comment (5aaee9) [reply pkgs/os-specific/linux/yt6801/default.nix]: Updated

**Review by FliegendeWurst — APPROVED** [review]

**Comment by bartvdbraak** [comment]
I had created almost the same addition with minor differences. This driver is used for these types of machines, of which I am using the first (TongFang GX4HRXL) personally:

- [TongFang GX4HRXL](https://laptopparts4less.frl/laptop-samenstellen/AMD-Ryzen-AI-laptop-samenstellen/TongFang-GX4HRXL-AMD-R7-8845HS-laptop-samenstellen)
- [TUXEDO InfinityBook Pro 14 - Gen9 - AMD](https://www.tuxedocomputers.com/en/TUXEDO-InfinityBook-Pro-14-Gen9-AMD#gallery)
- [SKIKK GREEN 6](https://www.skikk.eu/nl/laptops/green)
- [WootBook Y14](https://www.wootware.co.za/wootbook-y14-amd-ryzen-7-8845hs-5-10ghz-boost-8-core-14-2-8k-2880x1800-120hz-grey-laptop.html?srsltid=AfmBOoo-DEJkgRE9VpwDMccXLpBA-tOxc_A87b8O0mjfDDIihQCZQ3wS)

Let me know if there is anything you would like me to test.

File: pkgs/os-specific/linux/yt6801/default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      --replace-fail 'depmod $(shell uname -r)' "" \
+      --replace-fail 'modprobe $(KFILE)' ""
+  '';
+
+  makeFlags = [
+    "KERNELRELEASE=${kernel.modDirVersion}"
+    "KSRC_BASE="
+    "KSRC=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    "KDST=kernel/drivers/net/ethernet/motorcomm"
+    "INSTALL_MOD_PATH=$(out)"
```
Comment (pluiedev) [inline pkgs/os-specific/linux/yt6801/default.nix:37]:
```suggestion
    "INSTALL_MOD_PATH=${placeholder "out"}"
```

File: pkgs/os-specific/linux/yt6801/default.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    "ko_dir=${modDestDir}"
+    "ko_full=${modDestDir}/yt6801.ko.xz"
+  ];
+
+  meta = {
+    homepage = "https://www.motor-comm.com/product/ethernet-control-chip";
+    description = "YT6801 Gigabit PCIe Ethernet controller chip";
+    license = lib.licenses.gpl2;
```
Comment (pluiedev) [inline pkgs/os-specific/linux/yt6801/default.nix:45]: gpl2Only or gpl2Plus?

File: pkgs/os-specific/linux/yt6801/default.nix [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    maintainers = with lib.maintainers; [ indexyz ];
+    platforms = lib.platforms.linux;
```
Comment (pluiedev) [inline pkgs/os-specific/linux/yt6801/default.nix:47]:
```suggestion
    platforms = lib.platforms.linux;
    sourceProvenance = with lib.sourceTypes; [ binaryFirmware ];
```

File: pkgs/os-specific/linux/yt6801/default.nix [hunk 5]
```diff
@@ -0,0 +1,49 @@
+{
+  stdenv,
+  lib,
```
Comment (pluiedev) [inline pkgs/os-specific/linux/yt6801/default.nix:3]: Nit: lib should be placed above stdenv

File: pkgs/os-specific/linux/yt6801/default.nix [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+in
+stdenv.mkDerivation rec {
+  pname = "yt6801";
+
+  version = "1.0.29";
+  src = fetchzip {
+    stripRoot = false;
+    url = "https://www.motor-comm.com/Public/Uploads/uploadfile/files/20240812/yt6801-linux-driver-${version}.zip";
+    sha256 = "sha256-oz6CeOUN6QWKXxe3WUZljhGDTFArsknjzBuQ4IchGeU=";
```
Comment (pluiedev) [inline pkgs/os-specific/linux/yt6801/default.nix:20]: Also: are you sure that the date in the download link is the same for all versions? If not then it should be its own updatable variable
```suggestion
    url = "https://www.motor-comm.com/Public/Uploads/uploadfile/files/20240812/yt6801-linux-driver-${finalAttrs.version}.zip";
    hash = "sha256-oz6CeOUN6QWKXxe3WUZljhGDTFArsknjzBuQ4IchGeU=";
    stripRoot = false;
```

File: pkgs/os-specific/linux/yt6801/default.nix [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+}:
+
+let
+  modDestDir = "$(out)/lib/modules/${kernel.modDirVersion}/kernel/drivers/net/ethernet/motorcomm";
+
+in
+stdenv.mkDerivation rec {
+  pname = "yt6801";
+
+  version = "1.0.29";
```
Comment (pluiedev) [inline pkgs/os-specific/linux/yt6801/default.nix:16]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
  pname = "yt6801";
  version = "1.0.29";

```

**Review by pluiedev — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/yt6801/default.nix (hunk 4 above)
Comment (5aaee9) [reply pkgs/os-specific/linux/yt6801/default.nix:47]: I think it should be `fromSource` instand of `binaryFirmware`? It's don't contain blobs for firmware.

**Review by pluiedev — APPROVED** [review]

**Comment by bartvdbraak** [comment]
❤️❤️

**Pull request merged**